src/builtinHelpers.js

```diff
@@ -78,6 +78,11 @@ let section = function section(context, params, bodies, td) {
   let key = params.key.split('.');
   let val = td.get(context, key);
   let body, ctx;
+  Object.keys(params).forEach((key) => {
+    if (key !== 'key') {
+      td.helperContext.set(key, params[key]);
```

Comment:
The helper context is cleared out when the helper ends, unless you use `setGlobal`. Global values stay until rendering completes (maybe after - that's worth checking out).

The idea is that, if you have nested helpers, you have a stack of helper contexts. The inner helpers have access to the parent helpers' contexts, but when a helper exists, its context is destroyed.